src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/main/java/org/geoserver/security/oauth2/ValidatingOAuth2RestTemplate.java

```diff
@@ -109,7 +112,13 @@ private void validate(OAuth2AccessToken token) {
             String idToken = (String) maybeIdToken;
             setAsRequestAttribute(OpenIdConnectAuthenticationFilter.ID_TOKEN_VALUE, idToken);
             // among other things, this verifies the token
-            if (store != null) store.readAuthentication(idToken);
+            if (store != null) {
+                try {
+                    store.readAuthentication(idToken);
+                } catch (InvalidTokenException e) {
+                    LOGGER.warning("Failed to validate ID token: " + e.getMessage());
+                }
```

Comment:
What if the validation failed for something other than the kid verification? This seems to be ignoring all other sorts of validity issues?

Reply:
This is actually the fix, the KID verification is optional

Reply:
Is KID verification the only possible reason to get an InvalidTokenException? Or will we ignore also verification failures that should be considered?

Reply:
Most likely this is the reason, other reasons might be related to the impossibility to decrypt due to cypher algorythm. In any case the verification is optional for the protocol.
In order to intercept the specific one what we can do is to inspect the exception message probably even if, generally speaking, that shouldn't be a blocking error at all.

What about having an option on the GUI in order to enforce the velidation or not instead?

Reply:
I agree that sounds like a safer approach (let the admin decide whether to verify or not)

Reply:
Done